### ISSUE TITLE
Stay on (yank, indent, outdent, delete, etc) feature

### ISSUE THREAD

**Issue by davewasmer** [VSCodeVim/Vim#2397]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature Request

One of the nice things I like about Atom's popular vim plugin, vim-mode-plus, is the option to enable `stayOn{Yank,Delete,etc}`. These options "try to keep original cursor position as much as posible after operation". It is an intentional difference from pure vim, and is behind a configuration flag.

I've got a local copy of this plugin that mimics that behavior - would you be open to a PR that adds similar options to this plugin? Just wondering before I go through the effort to package up a neat PR for it.

**Comment by davewasmer** [comment]
For reference, the vim-mode-plus docs for this feature: https://github.com/t9md/atom-vim-mode-plus/wiki/DifferencesFromPureVim#config-vmp-stayontransformstring-stayonyank-stayondelete

**Comment by jpoon** [comment]
Absolutely, we'd be happy to take PRs. Take a look at the existing implementations for `Easymotion`, `Sneak`.

**Comment by trkoch** [comment]
I'd also like to see this feature in VSCodeVim. @davewasmer, have you made any progress? How can I help?

**Comment by davewasmer** [comment]
@trkoch see #2622

**Comment by jdubwelch** [comment]
Currently I can maintain the current cursor position by adding this to my settings. 

```
"vim.visualModeKeyBindingsNonRecursive": [
        {
            "before": [
                "y",
            ],
            "after": [ "y", "g", "v", "<ESC>" ]
        }
    ],
```

If you want to maintain the highlight, just remove the "<ESC>" at the end.

**Comment by DylanAndrews** [comment]
Did this ever actually get added. From what I can tell it didn't (no instance of `stayOnYank` in the source). Just wanted to check because it would be awesome to have it. Thanks!

**Comment by J-Fields** [comment]
No, this is not implemented. Would accept a PR if anyone wants to give it a shot.
